src/openrct2/ride/Ride.cpp

```diff
@@ -4579,6 +4579,41 @@ uint8_t ride_get_helix_sections(Ride* ride)
     return ride->special_track_elements & 0x1F;
 }
 
+bool Ride::HasElement(uint16_t element, bool useName) const
+{
+    bool moveSlowIt = true;
+    track_circuit_iterator it, slowIt;
+    int n = 0;
+    auto el = map_get_nth_element_at(stations[0].Start, n);
+    while (el == nullptr || el->GetType() != TileElementType::Track || el->AsTrack()->GetRideIndex() != id)
+    {
+        n++;
+        el = map_get_nth_element_at(stations[0].Start, n);
+    }
+    CoordsXYE start = CoordsXYE(stations[0].Start, el);
+    track_circuit_iterator_begin(&it, start);
+    slowIt = it;
+
+    while (track_circuit_iterator_next(&it))
+    {
+        auto trackType = it.current.element->AsTrack()->GetTrackType();
+        if (trackType == element || (useName && RideConfigurationStringIds[trackType] == RideConfigurationStringIds[element]))
```

Comment:
What is this check?? `RideConfigurationStringIds[trackType] == RideConfigurationStringIds[element]` ?? why would the stringId be what you compare. Compare the actual value.

Reply:
> don't ask me why. For example, Helix Up Large is in there 4 times

There are two types of helixes, used on different rides: one helix that builds in sections with 180 degrees of curve, one that builds in sections of 90 degrees curve. For example, the steel looper coaster builds with the 180 degree helixes, while the suspended swinging coaster builds 90 degree helix sections.

For each of those two types, you probably also have distinct track elements for left helix and right helix.

And that gives you 4 total types of Helix Large Up elements.